src/components/app-configure/auth/SocialAuth.vue

```diff
@@ -0,0 +1,172 @@
+<script lang="ts" setup>
+import { computed, type ComputedRef } from 'vue'
+import { useStore } from 'vuex'
+
+import CopyIcon from '@/assets/iconography/copy.svg'
+import LinkIcon from '@/assets/iconography/link.svg'
+import SettingCard from '@/components/app-configure/SettingCard.vue'
+import VSeperator from '@/components/lib/VSeperator/VSeperator.vue'
+import VStack from '@/components/lib/VStack/VStack.vue'
+import VTextField from '@/components/lib/VTextField/VTextField.vue'
+import {
+  socialLogins,
+  type SocialAuth,
+  type SocialAuthVerifier,
+  type SocialAuthField,
+} from '@/utils/constants'
+import copyToClipboard from '@/utils/copyToClipboard'
+
+const store = useStore()
+
+const socialAuthTypes: SocialAuthField[] = [
+  'clientId',
+  'clientSecret',
+  'redirectUri',
+]
+
+const authDetails: ComputedRef<SocialAuth[]> = computed(
+  () => store.getters.socialAuth
+)
+
+function getAuthIndex(
+  authDetailsList: SocialAuth[],
+  verifier: SocialAuthVerifier
+): number {
+  return authDetailsList.findIndex(
+    (authDetail: SocialAuth) => authDetail.verifier === verifier
+  )
+}
+
+function getClient(verifier: SocialAuthVerifier): SocialAuth | undefined {
+  const tempAuthDetails: SocialAuth[] = [...authDetails.value]
+  let existingAuthIndex = getAuthIndex(tempAuthDetails, verifier)
+  if (existingAuthIndex !== -1) {
+    return tempAuthDetails[existingAuthIndex]
+  }
+}
+
+function handleUpdate(
+  verifier: SocialAuthVerifier,
+  inputType: SocialAuthField,
+  value: string
+) {
+  let tempAuthDetails: SocialAuth[] = [...authDetails.value]
+  const existingAuthIndex = getAuthIndex(tempAuthDetails, verifier)
+  if (existingAuthIndex !== -1) {
+    const existingAuth = { ...tempAuthDetails[existingAuthIndex] }
+    const otherInputTypes: SocialAuthField[] = socialAuthTypes.filter(
+      (el) => el !== inputType
+    )
+    if (
+      !value &&
+      !existingAuth[otherInputTypes[0]] &&
+      !existingAuth[otherInputTypes[1]]
+    ) {
+      tempAuthDetails = tempAuthDetails.filter(
+        (authDetail) => authDetail.verifier !== existingAuth.verifier
+      )
```

Comment:
@ajithranka @karthik-durai , have written this logic to remove the auth type from the store if all it's fields are empty, i.e clientId, clientSecret, redirectUri. For cases where redirectUri and clientSecret is not required, this condition will remove auth only if clientId is empty

Reply:
Let's add the cross icon and clear all fields on click.